app/presenters/edition_presenter.rb

```diff
@@ -0,0 +1,120 @@
+module Presenters
+  class EditionPresenter
+    def initialize(edition, draft: false)
+      @edition = edition
+      @draft = draft
+    end
+
+    def for_content_store(payload_version)
+      present.except(:update_type).merge(payload_version: payload_version)
+    end
+
+    def for_message_queue(update_type)
+      present.merge(
+        update_type: update_type,
+        govuk_request_id: GdsApi::GovukHeaders.headers[:govuk_request_id],
+        links: unexpanded_links
+      )
+    end
+
+  private
+
+    attr_reader :draft, :edition
+
+    def unexpanded_links
+      Queries::LinkSetPresenter.new(
+        LinkSet.find_by(content_id: edition.content_id)
+      ).links
+    end
+
+    def present
+      edition.to_h
+        .except(*%i{updated_at created_at document_id content_store last_edited_at id state user_facing_version api_url web_url withdrawn api_path unpublishing_type}) # only intended to be used by publishing applications
```

Comment:
This is getting quite long now, might be worth extracting out into a constant at the top of the file with a name explaining what it is for. (I remember skipping past this line once before because I didn't know what it did when it was the exact thing I was looking for.)